### ISSUE TITLE
In Windows Desktop the server connection can be closed silently - can we have an alert?

### ISSUE THREAD

**Issue by Miles-Davies-HORIBA** [mattermost/desktop#608]
I confirm (by marking "x" in the [ ] below): 

- [x ] This is not a troubleshooting question. [Troubleshooting questions go here: http://www.mattermost.org/troubleshoot/](http://www.mattermost.org/troubleshoot/).
- [x ] This doesn't reproduce on web browsers (such as in Chrome). If it does, [issue reports go to the Mattermost Server repository](https://github.com/mattermost/platform/issues).
- [x ] I have read [contributing guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md). 

---

**Summary**

Implement an alert/notification if server connection is dropped
**Steps to reproduce**
Windows 7
Client 3.7.0
Server 3.10.0

The client returns to the server login screen silently at seemingly random

**Expected behavior**
If the client is disconnected/logged out from the server I would expect a notification/alert to indicate this
**Observed behavior**
There is no notification

**Possible fixes**
If the message alert were used with an exclamation mark rather than a count of unread messages when the client application is not logged in to any server that would be helpful. Currently our chat is infrequent and users can be logged out silently without realizing this. An alert would let them know they need to reconnect

**Comment by jasonblais** [comment]
@yuya-oc A push notification was added for Mattermost mobile apps when the session expires: https://mattermost.atlassian.net/browse/RN-17

Maybe we could consider something similar for the desktop app? What do you think?

**Comment by yuya-oc** [comment]
We might be able to set badges and send a desktop notification when the URL is `<server address>/login`.

**Comment by Miles-Davies-HORIBA** [comment]
Just having some visual clue that you are no longer logged in and will be getting no messages until you do would be great. At the moment some of the team don't respond because the desktop client has disconnected and they simply are not aware that they are no longer receiving notifications

**Comment by jasonblais** [comment]
Thanks! The badge may be more difficult given you might be connected to more than one server and have purposely logged out from one of them.

@yuya-oc Would the desktop notification be triggered whenever you land on the `<server address>/login` page? For instance after (intentionally) logging out?

**Comment by yuya-oc** [comment]
@jasonblais Yes. Probably I can't detect the timing of logging out.

About badges, surely it would be difficult when there are multiple servers. So how about using only tab badge? While logging out, of course there is no mention.

**Comment by jasonblais** [comment]
@yuya-oc Would it be feasible if this was a server-side change where a desktop notification is triggered whenever the user's session expires? We could then have an if condition to apply it for the app only, or have it also for browsers (people using the browser might have the same problem).

One concern with having the tab badge is that it wouldn't be obvious if the app is minimized. The user wouldn't know their session is expired until they actually opened the app (and either saw the login page or the tab badge)

**Comment by jasonblais** [comment]
@yuya-oc, thoughts on the above note?

**Comment by yuya-oc** [comment]
@jasonblais Sorry, I missed your response.

> Would it be feasible if this was a server-side change where a desktop notification is triggered whenever the user's session expires? We could then have an if condition to apply it for the app only, or have it also for browsers (people using the browser might have the same problem).

I haven't met though, does the webapp automatically moves to login page when user's session has been expired while the user is opening the webapp? If this is true, that's possible also on browser.

> One concern with having the tab badge is that it wouldn't be obvious if the app is minimized. The user wouldn't know their session is expired until they actually opened the app (and either saw the login page or the tab badge)

Surely, but even when the app is activated, the user can't know whether the hidden tabs' sessions are expired. So I think we should consider tab badges at least.

**Comment by jasonblais** [comment]
Yeah, the webapp automatically moves to the login page.

I created a design ticket for the product team to consider. I think the behaviour should be consistent between the webapp and the desktop app: https://mattermost.atlassian.net/browse/UX-85

I see your point about the badge. I have some worries about it, as it wouldn't work if the user has only connected to one server. But I included it as an option in the JIRA ticket.
